### ISSUE TITLE
Setters in raw JS type with varargs are allowed

### ISSUE THREAD

**Issue by gzm0** [scala-js/scala-js#1706]
The following compiles but probably shouldn't:

```scala
trait A extends js.Object {
  def a_=(x: Int*): Unit = js.native
}
```

The `@JSExport` equivalent does fail:

```
A method ending in _= will be exported as setter. But foo_= does not have the right signature to do so (single argument, unit return type).
```

**Comment by gzm0** [comment]
@sjrd, please confirm this.

**Comment by sjrd** [comment]
It should definitely not compile. The compiler wouldn't know what to do with a call site `x.a_=(a, b, c)`!

**Comment by gzm0** [comment]
I'm pretty sure it will set the `Seq` on the property...

Shall I make this a hard fail or should we make it warn?

**Comment by sjrd** [comment]
Hard fail. There's no reason not to.

**Comment by gzm0** [comment]
Ok

**Comment by sjrd** [comment]
Fixed in 7cacc307c3a9184ce317268f9489353bc8a862a4
